csrc/device_lower/analysis/predicate_elimination.cpp

```diff
@@ -1035,13 +1036,27 @@ Val* PredicateElimination::getInitValue(TensorView* tv) const {
 
 std::string PredicateElimination::toString() const {
   std::stringstream ss;
-  ss << "Tensors that do not need predication:";
+  VectorOfUniqueEntries<TensorView*> non_predicated_tvs;
   for (auto expr : non_predicated_exprs_) {
     for (auto out : expr->outputs()) {
-      NVF_ERROR(out->isA<TensorView>());
-      ss << " T" << out->name();
+      if (auto ti = dynamic_cast<kir::TensorIndex*>(out)) {
+        non_predicated_tvs.pushBack(ti->view());
+      } else if (auto tv = dynamic_cast<TensorView*>(out)) {
+        non_predicated_tvs.pushBack(tv);
+      } else {
+        NVF_ERROR(
+            false,
+            "Unexpected output ",
+            out->toString(),
```

Comment:
Why? I saw weird behavior for using pointers directly like suggested. Sometimes it prints the pointer address, but sometimes it prints the nice string as generated in `toString`. I believe the root cause is because by default, C++ already has `operator<<(std::ostream&, T*)` for pointers that print address, but we are defining our own `operator<<(std::ostream&, T*)` for our own IR node in some header. So depending on whether this header is included or not, the print of pointers will be different, and I feel this behavior confusing, that's why I always manually `toString`. Is this problem resolved? Do we have a general guideline on what we should prefer in our coding style?